src/javascript/class-scanner.ts

```diff
@@ -335,6 +347,221 @@ interface CustomElementDefinition {
+      return;
+    }
+
+    const left = node.expression.object;
+
+    // we only want `something.prototype.member`
+    if (!babel.isIdentifier(node.expression.property) ||
+        !babel.isMemberExpression(left) ||
+        getIdentifierName(left.property) !== 'prototype') {
+      return;
+    }
+
+    const cls = getIdentifierName(left.object);
+
+    if (!cls) {
+      return;
+    }
+
+    const annotation = getJSDocAnnotationForNode(node);
+
+    if (jsdoc.hasTag(annotation, 'function')) {
+      const method = this._createMethodFromExpression(node.expression.property.name,
+        node.expression, annotation);
+      if (method) {
+        this._addMethodToClass(cls, method);
+      }
+    } else {
+      const prop = this._createPropertyFromExpression(node.expression.property.name,
+        node.expression, annotation);
+      if (prop) {
+        this._addPropertyToClass(cls, prop);
+      }
+    }
+  }
+
+  private _createPropertyFromExpression(
+      name: string, node: babel.AssignmentExpression|babel.MemberExpression, jsdocAnn?: jsdoc.Annotation) {
+    let description;
+    let type;
+    let privacy: Privacy = 'public';
+    let readOnly = false;
+    const sourceRange = this._document.sourceRangeForNode(node)!;
+    const warnings: Warning[] = [];
+
+    if (jsdocAnn) {
+      const typeTag = jsdoc.getTag(jsdocAnn, 'type');
+      if (typeTag && typeTag.type) {
+        type = doctrine.type.stringify(typeTag.type);
+      }
+      description = getDescription(jsdocAnn);
+      privacy = getOrInferPrivacy(name, jsdocAnn);
+      readOnly = jsdoc.hasTag(jsdocAnn, 'readonly');
+    }
+
+    if (babel.isAssignmentExpression(node)) {
+      if (!type) {
+        const detectedType = closureType(node.right, sourceRange, this._document);
+        if (detectedType instanceof Warning) {
+          warnings.push(detectedType);
+          type = 'Function';
+        } else {
+          type = detectedType;
+        }
+      }
+    }
+
+    return {
+      name,
+      astNode: node,
+      type,
+      jsdoc: jsdocAnn,
+      sourceRange,
+      description,
+      privacy,
+      warnings,
+      readOnly,
+    };
+  }
+
+  private _createMethodFromExpression(
+      name: string, node: babel.FunctionExpression|babel.MemberExpression, jsdocAnn?: jsdoc.Annotation) {
+    let description;
+    let type;
+    let privacy: Privacy = 'public';
+    let ret;
+    const params = new Map<string, MethodParam>();
+
+    if (jsdocAnn) {
+      description = getDescription(jsdocAnn);
+      privacy = getOrInferPrivacy(name, jsdocAnn);
+      ret = getReturnFromAnnotation(jsdocAnn);
+      type = ret ? ret.type : undefined;
+
+      if (babel.isFunctionExpression(node)) {
+        (node.params || []).forEach((nodeParam) => {
+          const param = toMethodParam(nodeParam, jsdocAnn);
+          params.set(param.name, param);
+        });
+      } else {
+        for (const tag of (jsdocAnn.tags || [])) {
```

Comment:
We should run this even if it's a function expression. The documentation may have more info on the parameters, like type and description.

Reply:
`toMethodParam` does a very similar bit of logic already. so we do pick up types and descriptions from the annotations through that